docs/src/modules/ROOT/pages/upgrading-timefold-solver/backwards-compatibility.adoc

```diff
@@ -5,50 +5,39 @@
 
 Timefold Solver separates its API from its implementation:
 
-* **Public API**: All classes in the following package namespaces are 100% *backwards compatible* in future releases,
-especially minor and hotfix releases:
+* **Public API**: All classes under these `api` and `config` namespaces are 100% *backwards compatible* in future minor and hotfix releases:
 ** `ai.timefold.solver.core.api`
 ** `ai.timefold.solver.benchmark.api`
 ** `ai.timefold.solver.test.api`
-** `ai.timefold.solver...api*`
-* **Configuration**: The solver configuration is backwards compatible for all elements,
-except for elements that require the use of non-public API classes.
-The solver configuration is defined by the classes in the following package namespaces:
+** `ai.timefold.solver.*.api`
```

Comment:
Ever since we split up the config and factory classes, during the XSteam to Jaxb refactor, this complexity is not longer relevant and can be removed. The config classes are simply backwards compatible.

Especially nowadays, as people use the Java config far more than the XML config.